### ISSUE TITLE
Syntax errors in build.rs

### ISSUE THREAD

**Issue by Silur** [maidsafe/rust_sodium#16]
When cargo tries to build the `rust_sodium_sys` package, it runs into several syntactical errors with make, cc, ar:

```
>cargo build
.......
ar: `u' modifier ignored since `D' is the default (see `U')
cc: error: 0: No such file or directory
cc: error: RESERVED_FUNCTION_POINTERS=8: No such file or directory
cc: error: unrecognized command line option '--memory-init-file'; did you mean '--warn-init-self'?
cc: error: unrecognized command line option '--pre-js'; did you mean '--prefix'?
make[3]: *** [Makefile:1194: aead_aes256gcm.js] Error 1
make[3]: *** Waiting for unfinished jobs....
cc: error: 0: No such file or directory
cc: error: RESERVED_FUNCTION_POINTERS=8: No such file or directory
cc: error: unrecognized command line option '--memory-init-file'; did you mean '--warn-init-self'?
cc: error: unrecognized command line option '--pre-js'; did you mean '--prefix'?
make[3]: *** [Makefile:1198: aead_chacha20poly1305.js] Error 1
make[2]: *** [Makefile:2176: check-am] Error 2
make[1]: *** [Makefile:394: check-recursive] Error 1
make: *** [Makefile:505: check-recursive] Error 1

', github.com-1ecc6299db9ec823/rust_sodium-sys-0.1.1/build.rs:233
```

**Comment by Fraser999** [comment]
Hi there,

Thanks for the report.  To let us reproduce this, could you tell us your build environment please (OS and version, rustc version)?

**Comment by Silur** [comment]
of course sorry, I'm using 
Linux 4.7.4-1-ARCH x86_64
rustc 1.11.0
cargo 0.12.0 (6b98d1f 2016-07-04)

**Comment by Fraser999** [comment]
Thanks for that.

We're not building or running CI tests on Arch Linux currently, so it may be a while until someone gets a chance to look at this properly - sorry.

In the meantime, I'll leave the issue open.

**Comment by afck** [comment]
We added some more output in the case of build errors: #18 
Could you please try again with latest master?
`rust_sodium_sys/build.rs` tries to build libsodium and seems to fail in the `./configure` step.

**Comment by chpio** [comment]
I think it's testing for emscripten and setting flags if it's present. But then calls cc instead of emscripten with emscripten flags. run `unset EMSCRIPTEN` as a workaround

*./configure*:
```bash
# ...
TEST_LDFLAGS=''
if test "x$EMSCRIPTEN" != "x"; then :

  EXEEXT=.js
  TEST_LDFLAGS='--memory-init-file 0 --pre-js pre.js.inc -s RESERVED_FUNCTION_POINTERS=8'

fi
# ...
```

**Comment by oconnor663** [comment]
`unset EMSCRIPTEN` did the trick for me, thank you. Or in my case, since I didn't need it, I just uninstalled the emscripten package (and restarted my machine).

**Comment by chpio** [comment]
fixed in https://github.com/jedisct1/libsodium/issues/544

**Comment by Fraser999** [comment]
Closed via #60
